src/components/calls-page/calls-page.tsx

```diff
@@ -70,6 +84,61 @@ export const CallsPage = () => {
     initialHotkey: "p",
   });
 
+  const callActionHandlers = useRef<Record<string, Record<string, () => void>>>(
+    {}
+  );
+  const callIndexMap = useRef<Record<number, string>>({});
+
+  useEffect(() => {
+    callIndexMap.current = {};
+    Object.keys(calls).forEach((callId, i) => {
+      callIndexMap.current[i + 1] = callId;
+    });
+  }, [calls]);
+
+  const muteToggleTimeoutRef = useRef<number | null>(null);
+
+  const handleToggleGlobalMute = () => {
+    if (muteToggleTimeoutRef.current !== null) return;
+
+    setIsMasterInputMuted((prev) => {
+      const newMuteState = !prev;
+
+      setTimeout(() => {
+        sendCallsStateUpdate();
+      }, 0);
+
+      return newMuteState;
+    });
+
+    setIsSettingGlobalMute(true);
+
+    muteToggleTimeoutRef.current = window.setTimeout(() => {
+      muteToggleTimeoutRef.current = null;
+    }, 300);
+
+    window.setTimeout(() => {
+      setIsSettingGlobalMute(false);
+    }, 1000);
+  };
```

Comment:
Nice rewrites! Maybe this could be in a callback-hook as well? 
Along with `const [isSettingGlobalMute, setIsSettingGlobalMute] =
    useState<boolean>(false);`